### ISSUE TITLE
CLI ~v3.92.0 broke using ESC secret as auth_token for GCP->Slack NotificationChannel

### ISSUE THREAD

**Issue by dcox-at-bgrove** [pulumi/pulumi#14513]
We have a yaml-based Pulumi program, which largely defines Google Cloud (gcp) resources, including
```
  # a notification channel used in subsequent alert policy
  slack-alert-channel:
    type: gcp:monitoring:NotificationChannel
    properties:
      displayName: Slack ${esc_notification_slack_channel}
      description: Channel for notifying engineers in Slack
      project: ${gcp_project}
      enabled: true
      forceDelete: false # ?
      type: slack
      labels: # following are specific to type slack
        channel_name: ${esc_notification_slack_channel}
        auth_token: "${esc_slack_auth_token}"
```
The `auth_token` on the last line is sourced from Pulumi ESC Environments such as our `agent-services-dev` (or `-prod`), which import our `agent-services`, which defines `esc_slack_auth_token` as a valid slack token that works for the above NotificationChannel resource.  The token itself still works (if hardcoded in place of `${esc_slack_auth_token}` above).
```
> pulumi env get bgrove/agent-services    

   Value 
  
    {
      "alerting": {
        "opsgenie_agent_team_api_key": "[secret]",    
        "slack_auth_token": "[secret]"
      },
      "pulumiConfig": {
        "esc_opsgenie_agent_team_api_key": "[secret]",
        "esc_slack_auth_token": "[secret]"
      }
    }

   Definition

    values:
      alerting:
        opsgenie_agent_team_api_key:
          fn::secret:
            ciphertext: ZXNje...

        slack_auth_token:
          fn::secret:
            ciphertext: ZXNje...

      pulumiConfig:
        esc_opsgenie_agent_team_api_key: ${alerting.opsgenie_agent_team_api_key}
        esc_slack_auth_token: ${alerting.slack_auth_token}
```
This has been working since I implemented it about 3 weeks ago.  I developed and tested it with Pulumi CLI v3.88.1, and since then we've been using it in CircleCI to deploy, including to production, via their Pulumi Orb, which by default uses the latest Pulumi CLI.

Recently the CircleCI deploy job started failing in `pulumi update`:
```
#!/bin/bash -eo pipefail
pulumi update --yes --stack bgrove/us-dev --cwd data_files/deployment/agentapi/pulumi

Previewing update (us-dev)

View Live: https://app.pulumi.com/bgrove/*********.agentapi/us-dev/previews/b1695346-e13f-4f85-b777-362dd2d1eb3e

[resource plugin gcp-6.67.1] installing
@ Previewing update.....

    pulumi:pulumi:Stack *********.agentapi-us-dev running 
@ Previewing update.....
    gcp:storage:Bucket airs-bucket-non-regional  
    gcp:storage:Bucket airs-bucket
      ⋮  
    gcp:pubsub:Topic heartbeat-topic  
    gcp:pubsub:Topic trace-logging-topic  
    gcp:logging:Metric time-drift-metric  
    gcp:logging:Metric cross-account-metric  
    gcp:logging:Metric unknown-account-metric  
    gcp:logging:Metric agentapi-policy-ratelimit_rate_limit_exceed_preview  
    gcp:logging:Metric logged-errors-metric  
    gcp:monitoring:NotificationChannel slack-alert-channel  error: gcp:monitoring/notificationChannel:NotificationChannel resource 'slack-alert-channel' has a problem: '' expected type 'string', got unconvertible type 'map[string]interface {}', value: 'map[]'. Examine values at 'slack-alert-channel.labels["auth_token"]'.
    pulumi:pulumi:Stack *********.agentapi-us-dev  
    gcp:monitoring:NotificationChannel slack-alert-channel  1 error
Diagnostics:
  gcp:monitoring:NotificationChannel (slack-alert-channel):
    error: gcp:monitoring/notificationChannel:NotificationChannel resource 'slack-alert-channel' has a problem: '' expected type 'string', got unconvertible type 'map[string]interface {}', value: 'map[]'. Examine values at 'slack-alert-channel.labels["auth_token"]'.
```
So `slack-alert-channel.labels["auth_token"]` expects a value of type string, but the ESC aspect of the CLI is providing something else?

I noticed that prior to the above, the CircleCI job installed Pulumi CLI v3.92.0.  I could not reproduce the error on my development machine until I updated the CLI from v3.88.1 to v3.92.0, and I was able to avoid the error by downgrading the CLI back to v3.88.1.

I changed the `NotificationChannel` definition to use `sensitiveLabels` to pass the `authToken`:
```
  slack-alert-channel:
    type: gcp:monitoring:NotificationChannel
    properties:
      displayName: Slack ${esc_notification_slack_channel}
      description: Channel for notifying engineers in Slack
      project: ${gcp_project}
      enabled: true
      forceDelete: false # ?
      type: slack
      labels: # array of LabelDescriptor, specific to type slack
        channel_name: ${esc_notification_slack_channel}
      sensitiveLabels:
        authToken: "${esc_slack_auth_token}"
```
Under Pulumi CLI v3.92.0, this also encounters a type error:
```
  gcp:monitoring:NotificationChannel (slack-alert-channel):
    error: gcp:monitoring/notificationChannel:NotificationChannel resource 'slack-alert-channel' has a problem: Attribute must be a single value, not a map. Examine values at 'slack-alert-channel.sensitiveLabels.authToken'.
```

**Comment by dcox-at-bgrove** [comment]
@pgavlin Defining output `esc_slack_auth_token` as you have it seems to suppress that output item.  Note I added `frob` under `config` and `outputs`, and only that is output successfully:

`Pulumi.default.yaml`
```
environment:
  - agent-services-dev

config:
  static-secrets-test:frob: bob
```
`Pulumi.yaml`
```
name: static-secrets-test
runtime: yaml
config:
  esc_slack_auth_token:
    type: string
  frob:
    type: string
outputs:
  esc_slack_auth_token: ${esc_slack_auth_token}
  frob: ${frob}
```
`> pulumi up --yes --skip-preview --show-config`
```
Updating (default)

View in Browser (Ctrl+O): https://app.pulumi.com/bgrove/static-secrets-test/default/updates/11

     Type                 Name                         Status     Info
     pulumi:pulumi:Stack  static-secrets-test-default             1 message
Diagnostics:
  pulumi:pulumi:Stack (static-secrets-test-default):
    Configuration:
        static-secrets-test:esc_notification_slack_channel: #agent-alerts-dev
        static-secrets-test:esc_oncall_notification_enabled: false
        static-secrets-test:esc_opsgenie_agent_team_api_key: [secret]
        static-secrets-test:esc_slack_auth_token: [secret]
        static-secrets-test:frob: bob

Outputs:
    frob                : "bob"

Resources:
    1 unchanged

Duration: 1s
```

Note if I change the `esc_slack_auth_token` output to a string that includes interpolation of `${esc_slack_auth_token}`, then there is an output item for that:
```
name: static-secrets-test
runtime: yaml
config:
  esc_slack_auth_token:
    type: string
  frob:
    type: string
outputs:
  esc_slack_auth_token: "--- ${esc_slack_auth_token} ---"
  frob: ${frob}
```
`> pulumi up --yes --skip-preview --show-config`
```
Updating (default)

View in Browser (Ctrl+O): https://app.pulumi.com/bgrove/static-secrets-test/default/updates/16

     Type                 Name                         Status     Info
     pulumi:pulumi:Stack  static-secrets-test-default             1 message
Diagnostics:
  pulumi:pulumi:Stack (static-secrets-test-default):
    Configuration:
        static-secrets-test:esc_notification_slack_channel: #agent-alerts-dev
        static-secrets-test:esc_oncall_notification_enabled: false
        static-secrets-test:esc_opsgenie_agent_team_api_key: [secret]
        static-secrets-test:esc_slack_auth_token: [secret]
        static-secrets-test:frob: bob

Outputs:
    esc_slack_auth_token: "--- &{{[secret]}} ---"
    frob                : "bob"

Resources:
    1 unchanged

Duration: 1s
```

**Comment by pgavlin** [comment]
Okay, I think I have a handle on this.

`pulumi` 3.90.0 and `pulumi-yaml` 1.4.0 introduced a feature for better handling of typed configuration. I believe that this feature unintentionally broke handling of secret config. I'm looking into it more closely now.

**Comment by pgavlin** [comment]
https://github.com/pulumi/pulumi-yaml/issues/518

**Comment by pgavlin** [comment]
I think that this should be fixed as of `pulumi-yaml` 1.4.1.

https://github.com/pulumi/pulumi/pull/14542
